src/VocabularyManager.Api/Controllers/WordListController.cs

```diff
@@ -0,0 +1,140 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using VocabularyManager.Core.Entities;
+using VocabularyManager.Core.Interfaces;
+using VocabularyManager.Infrastructure.Validators;
+using VocabularyManager.Infrastructure.Exceptions;
+using VocabularyManager.UseCases.DTOs;
+using FluentValidation;
+
+namespace VocabularyManager.Api.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class WordListController : ControllerBase
+    {
+        private readonly IWordListService _wordListService;
+        private readonly IWordParser<Word> _oxfordDictionaryParser;
+        private readonly AbstractValidator<WordListDTO> _wordListValidator;
+        public WordListController(
+            IWordListService wordListService,
+            IWordParser<Word> oxfordDictionaryParser,
+            AbstractValidator<WordListDTO> wordListValidator)
+        {
+            _wordListService = wordListService;
+            _oxfordDictionaryParser = oxfordDictionaryParser;
+            _wordListValidator = wordListValidator;
+        }
+
+        [HttpPost("getwordlist")]
+        public async Task<IActionResult> GetWordListByUrlAsync([FromBody] string url)
+        {
+            try
+            {
+                IEnumerable<Word> wordList = await _oxfordDictionaryParser
+                    .GetWordListByLinkAsync(url, OxfordDictionaryValidator.IsWordObjectDataValid);
```

Comment:
- Infrastrutcure should implement interfaces from `Application` (UseCase in your case) layer. The only thing that API needs from infrastructure is to know implementation of interfaces.
- Why passing validation result instead of stopping execution if model its not valid?